TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/snapshot.py

```diff
@@ -38,55 +41,209 @@ def status(self, *args, **kwargs) -> CommandResult:
+            "--source-host", self.source_cluster.endpoint
         ]
-
         if self.source_cluster.allow_insecure:
             command.append("--source-insecure")
-        if self.target_cluster.allow_insecure:
-            command.append("--target-insecure")
+        if not wait:
+            command.append("--no-wait")
+        if max_snapshot_rate_mb_per_node is not None:
+            command.extend(["--max-snapshot-rate-mb-per-node",
+                            str(max_snapshot_rate_mb_per_node)])
 
         logger.info(f"Creating snapshot with command: {' '.join(command)}")
+
         try:
-            # Pass None to stdout and stderr to not capture output and show in terminal
             subprocess.run(command, stdout=None, stderr=None, text=True, check=True)
-            logger.info(f"Snapshot {self.config['snapshot_name']} created successfully")
-            return CommandResult(success=True, value=f"Snapshot {self.config['snapshot_name']} created successfully")
+            message = f"Snapshot {self.snapshot_name} created successfully"
+            logger.info(message)
+            return CommandResult(success=True, value=message)
         except subprocess.CalledProcessError as e:
-            logger.error(f"Failed to create snapshot: {str(e)}")
-            return CommandResult(success=False, value=f"Failed to create snapshot: {str(e)}")
+            message = f"Failed to create snapshot: {str(e)}"
+            logger.error(message)
+            return CommandResult(success=False, value=message)
 
-    def status(self, *args, **kwargs) -> CommandResult:
-        return CommandResult(success=False, value="Command not implemented")
+    def status(self, *args, **kwargs) -> Tuple[SnapshotStatus, str]:
+        deep_check = kwargs.get('deep_check', False)
+        if deep_check:
+            return get_snapshot_status_full(self.source_cluster, self.snapshot_name)
+        return get_snapshot_status(self.source_cluster, self.snapshot_name)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Elasticsearch snapshot status checker.")
+    parser.add_argument("--endpoint", help="Elasticsearch endpoint.", required=True)
+    parser.add_argument("--username", help="Cluster username.", default=None)
+    parser.add_argument("--password", help="Cluster password.", default=None)
+    parser.add_argument("--no-auth", action='store_true', help="Flag to provide no auth in requests.")
+    parser.add_argument("--debug", action='store_true', help="Enable debug logging.")
+    parser.add_argument("--detailed", action='store_true', help="Always get detailed status for completed snapshots.")
+    return parser.parse_args()
+
+
+class ClusterSnapshotState(Enum):
+    SUCCESS = "SUCCESS"
+    FAILED = "FAILED"
+    PARTIAL = "PARTIAL"
+    STARTED = "STARTED"
+    IN_PROGRESS = "IN_PROGRESS"
+
+
+def convert_snapshot_state_to_status(snapshot_state: str) -> Tuple[SnapshotStatus, str]:
```

Comment:
I'm not sure I understand why we should have State & Status and what the difference is. Should we just use these States throughout?

Reply:
State is the cluster response enum whereas Status is what we define. I had originally defined it when we were also storing status in a separate doc.

This helps us maintain a consistent Status despite state renaming across ES/OS versions

Reply:
This doesn't feel right, we are embedding version handling deep in the console.  I understand getting this in position for the demo, but this blends the responsibilities of the system quite a bit - it is going to make our codebase hard to maintain very quickly.

@AndreKurait What do you think of this, how should we capture the responsibility questions?

Reply:
Moved this to just return the state string that the cluster responds with